### PR TITLE
(deprecated) Enable users to create dynamic groups using language and created_on

### PR DESCRIPTION
**Pull request by dodobas**
Closes rapidpro/rapidpro#768

### CODE REVIEW

File: temba/contacts/search/parser.py [hunk 1]
```diff
@@ -278,12 +278,52 @@ def evaluate(self, contact_json, prop_map):
...
+
+            elif field == "created_on":
+                query_value = str_to_datetime(self.value, org.timezone, org.get_dayfirst(), fill_time=False)
+                if not query_value:
+                    raise SearchException(_(f"Unable to parse the date '{self.value}'"))
+
+                datetime_value = contact_json.get("created_on")
+                contact_value = str_to_datetime(datetime_value, org.timezone)
+
+                utc_range = date_to_utc_range(query_value.date(), org)
```
Comment (rowanseymour) [inline temba/contacts/search/parser.py]: since this is how we compare datetime field values feels like we can avoid repetition with a utility method

**Comment by nicpottier** [comment]
How are we keeping track of the `query_fields` for these two fields? That's used to figure out which groups need to be reevaluated isn't it?

File: temba/contacts/tests.py [hunk 2]
```diff
@@ -347,7 +347,7 @@ def test_evaluate_dynamic_groups_from_flow(self):
                 ContactField.get_or_create(self.org, self.admin, key, value_type=Value.TYPE_NUMBER)
                 ContactGroup.create_dynamic(self.org, self.admin, "Group %s" % (key), "(%s > 10)" % key)
 
-        with QueryTracker(assert_query_count=110, stack_count=16, skip_unique_queries=False):
+        with QueryTracker(assert_query_count=126, stack_count=16, skip_unique_queries=False):
```
Comment (dodobas) [inline temba/contacts/tests.py]: this increased because we are reevaluating dynamic groups

File: temba/api/v2/serializers.py [hunk 3]
```diff
@@ -525,10 +531,11 @@ def save(self):
             if "name" in self.validated_data and name != self.instance.name:
                 self.instance.name = name
                 changed.append("name")
+
             if "language" in self.validated_data and language != self.instance.language:
-                self.instance.language = language
-                changed.append("language")
+                self.instance.set_language(language)
 
+            # TODO: urns cannot be None because it is required field, so this will be always called and consequently reevalute dynamic groups
```
Comment (dodobas) [inline temba/api/v2/serializers.py]: the API and Contact.update (https://github.com/nyaruka/rapidpro/pull/2014/files#diff-95d38a65240a92d3ed796ac2c6b177a2R1373) are similar in a way that both of them will eventually call `update_urns` which will reevaluate dynamic groups
so in some cases we will reevaluate DG twice, if language has been changed and then later when we call `update_urns`

File: temba/contacts/models.py [hunk 4]
```diff
@@ -956,6 +959,18 @@ def set_field(self, user, key, value, label=None, importing=False):
         if has_changed and not importing:
             self.handle_update(field=field)
 
+    def set_language(self, language: Optional[str]):
```
Comment (rowanseymour) [inline temba/contacts/models.py]: As usual I wish we'd discuss things before adding new syntax 😄, but in this case I think we really should start using these type hints... unless anyone else has any objections?

Comment (dodobas) [reply temba/contacts/models.py]: How can you discuss things before you know how those 'look'?, you should be thanking me for creating a nice point of discussion :smile:

**Comment by dodobas** [comment]
superseded by #2056

**Pull request closed without merging**